### PR TITLE
perf(imgbtn): add img header cache to reduces file access

### PR DESCRIPTION
**Pull request by mumuge1**
### Description of the feature or fix

The current `lv_imgbtn` only saves the img src pointer. Due to frequent dimensional measurements, `lv_img_decoder_get_info` are called frequently, which can lead to a loss of performance.
Header caching needs to be added to reduce file access.

test code:
```c
#define ICON_EMOJI_STROE_ADD_NOR "/png/add_nor.png"
#define ICON_EMOJI_STROE_ADD_PRE "/png/add_pressed.png"
void imgbtn_test(void)
{
    lv_obj_t* emoji_btn = lv_imgbtn_create(lv_scr_act());
    lv_obj_set_size(emoji_btn, 40, 40);
    lv_imgbtn_set_state(emoji_btn, LV_IMGBTN_STATE_RELEASED);
    lv_imgbtn_set_src(emoji_btn, LV_IMGBTN_STATE_RELEASED, NULL, ICON_EMOJI_STROE_ADD_NOR, NULL);
    lv_imgbtn_set_src(emoji_btn, LV_IMGBTN_STATE_PRESSED, NULL, ICON_EMOJI_STROE_ADD_PRE, NULL);
    lv_obj_center(emoji_btn);
}
```

### Checkpoints
- [x] Run `code-format.py` from the scripts folder. [astyle](http://astyle.sourceforge.net/install.html) needs to be installed.
- [ ] Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed
- [ ] Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant. 
- [ ] Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- [ ] If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/release/v8.3/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/release/v8.3/Kconfig).
 
Be sure the following conventions are followed:
- [x] Follow the [Styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [x] Prefer `enum`s instead of macros. If inevitable to use `define`s export them with `LV_EXPORT_CONST_INT(defined_value)` right after the `define`.
- [x] In function arguments prefer `type name[]` declaration for array parameters instead of `type * name`
- [x] Use typed pointers instead of `void *` pointers
- [x] Do not `malloc` into a static or global variables. Instead declare the variable in `LV_ITERATE_ROOTS` list in [`lv_gc.h`](https://github.com/lvgl/lvgl/blob/master/src/misc/lv_gc.h) and mark the variable with `GC_ROOT(variable)` when it's used. See a detaild description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#memory-management).
- [x] Widget constructor must follow the `lv_<widget_name>_create(lv_obj_t * parent)` pattern.
- [x] Widget members function must start with `lv_<modul_name>` and should receive `lv_obj_t *` as first argument which is a pointer to widget object itself.  
- [x] `struct`s should be used via an API and not modified directly via their elements.
- [x] `struct` APIs should follow the widgets' conventions. That is to receive a pointer to the `struct` as the first argument, and the prefix of the `struct` name should be used as the prefix of the function name too (e.g.  `lv_disp_set_default(lv_disp_t * disp)`)
- [x] Functions and `struct`s which are not part of the public API must begin with underscore in order to mark them as "private".
- [x] Arguments must be named in H files too.
- [x] To register and use callbacks one of the followings needs to be followed (see a detaild description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#callbacks)): 
  - For both the registration function and the callback pass a pointer to a `struct` as the first argument. The `struct` must contain `void * user_data` field.
  - The last argument of the registration function must be `void * user_data` and the same `user_data` needs to be passed as the last argument of the callback.
  - Callback types not following these conventions should end with `xcb_t`.

### CODE REVIEW

File: src/widgets/imgbtn/lv_imgbtn.h [hunk 1]
```diff
@@ -30,16 +30,20 @@ typedef enum {
     _LV_IMGBTN_STATE_NUM,
 } lv_imgbtn_state_t;
 
+typedef struct {
+    const void * img_src;
+    lv_img_header_t header;
+} lv_imgbtn_src_pair_t;
```
Comment (kisvegabor) [inline src/widgets/imgbtn/lv_imgbtn.h]: Why is it called `pair`? What about `lv_imgbtn_src_info_t`?

**Review by kisvegabor — COMMENTED** [review]
Hi,

It looks good and makes sense. I have only one comment.

File: src/widgets/imgbtn/lv_imgbtn.h (hunk 1 above)
Comment (mumuge1) [reply src/widgets/imgbtn/lv_imgbtn.h]: Because I think they come in pairs

Comment (kisvegabor) [reply src/widgets/imgbtn/lv_imgbtn.h]: I think this name is not so flexible. If for whatever reason we need to add a new member there the name of the type needed to be changed too.

Comment (FASTSHIFT) [reply src/widgets/imgbtn/lv_imgbtn.h]: done

**Comment by kisvegabor** [comment]
Thank you!

**Pull request merged**